govultr.go

```diff
@@ -178,41 +184,46 @@ func (c *Client) NewRequest(ctx context.Context, method, uri string, body url.Va
 // have their own implements of unmarshal.
 func (c *Client) DoWithContext(ctx context.Context, r *http.Request, data interface{}) error {
 
-	// Sleep this call
+	// Initial sleep on the call
 	time.Sleep(c.RateLimit)
 
-	req := r.WithContext(ctx)
-	res, err := c.client.Do(req)
+	var body []byte
+	for tryCount := 1; tryCount <= c.RetryLimit; tryCount++ {
```

Comment:
It might be better for the request to be split into a separate private function so the defer for the closing the response body actually gets called after each request instead of after all the tried are exceeded. This would also make it easier to do testing by splitting the retry code from the request itself.